docs/en/week10/10-1.md

```diff
@@ -0,0 +1,271 @@
+
+**Disadvantages of SimCLR:**
+1. Large batch size required
+2. Large number of GPUs required
+
+**How do we solve the compute problem in the SimcLR?**
+
+A **Memory bank** can be used to maintain a momentum of activations across all features. With each forward pass of an image, the memory bank features are updated using the new embeddings. These memory bank features can then be used as negatives for contrastive loss. 
+
+#### Advantages of Memory bank:
+- Compute efficient. Requires one Forward Pass to compute the embeddings.
+
+#### Drawbacks of Memory bank:
+- Not online. Features become stale very fast.
+- Requires large amount of GPU RAM.
+- Hard to scale storage to millions of images.
+
+## MoCo
+
+MoCo, or, Momentum Contrast, is a contrastive method that utilizes a memory bank to maintain a momentum of activations. It works by using 2 encoders. The $f_{\theta}$ encoder is the encoder that needs to be learnt. $f_{\theta_{EMA}}$ encoder maintains an exponential moving average of $f_{\theta}$.
+
+<center>
+<img src="{{site.baseurl}}/images/week10/10-1/moco.png" style="width: 600px; background-color:#DCDCDC;" /><br>
+Figure 7: Architecture of MoCo
+</center>
+
+During Forward Pass, each sample is forwarded through both encoders. Some of the embeddings from $f_{\theta_{EMA}}$ are kept aside for use as negative embeddings. For the algorithm, original embeddings comes from $f_{\theta}$, positive embeddings from $f_{\theta_{EMA}}$ and negative embeddings come from the set of stored embeddings.
+
+#### Advantages of MoCo:
+- Can easily be scaled in terms of memory usage as the full dataset need not be stored.
+- Is an online solution, as the moving average is continuously updated.
+ 
+#### Drawbacks of MoCo:
+- Two Forward Passes are required, one through $f_{\theta}$ and another through $f_{\theta_{EMA}}$.
+- Extra memory required for parameters/stored features.
+
+
+## Clustering methods
+**How CL and clustering are related to one another?**
+
+In Constrastive Learning, each sample has corresponding positives and negatives, and the aim of the learning task is to try and bring together positive embeddings, while pushing apart negative embeddings. Essentially we are creating groups within the feature space.
+
+Clustering, on the other hand, is a more direct way to achieve grouping, as it naturally creates groups in the feature space.
+
+<center>
+<img src="{{site.baseurl}}/images/week10/10-1/contrastive-learning.png" style="width: 350px; background-color:#DCDCDC;" />
+<img src="{{site.baseurl}}/images/week10/10-1/clustering.png" style="width: 250px; background-color:#DCDCDC;" /><br>
+Figure 8: Difference between Contrastive learning (Left) and Clustering (Right)
+</center>
+
+
+## SwAV
+The SwAV (Swapping Assignments between Views) algorithm is an online clustering method. The idea here is to maximize the similarity of a given image $I$ and augmentation of that image $augment(I)$ and, in doing so, ensure they belong to the same cluster.
```

Comment:
Please, use `\text{}` for **ALL** non-standard function names, here and everywhere else.